be/ut/helper.c

```diff
@@ -398,6 +398,60 @@ M0_INTERNAL int m0_be_ut_backend_init_cfg(struct m0_be_ut_backend *ut_be,
 	return rc;
 }
 
+
+M0_INTERNAL int
+m0_be_ut_backend_init_cfg_create(struct m0_be_ut_backend *ut_be,
+				 const struct m0_be_domain_cfg *cfg,
+				 bool mkfs)
+{
+	struct m0_be_domain     *dom;
+	struct m0_be_domain_cfg *c;
+	int                      rc = 0;
+
+	m0_bob_init(&m0_ut_be_backend_bobtype, ut_be);
+
+	ut_be->but_sm_groups_unlocked = false;
+	m0_mutex_init(&ut_be->but_sgt_lock);
+
+	if (cfg == NULL)
+		m0_be_ut_backend_cfg_default(&ut_be->but_dom_cfg);
+	else
+		/*
+		 * Make a copy of `cfg': it can be an automatic variable,
+		 * which scope we should not rely on.
+		 */
+		ut_be->but_dom_cfg = *cfg;
+
+	c = &ut_be->but_dom_cfg;
+	dom = &ut_be->but_dom;
+
+	/* Create reqh, if necessary. */
+	if (c->bc_engine.bec_reqh == NULL)
+		m0_be_ut_reqh_create(&c->bc_engine.bec_reqh);
+
+	/* Use m0_be_ut_backend's stob domain location, if possible. */
+	if (ut_be->but_stob_domain_location != NULL)
+		c->bc_stob_domain_location = ut_be->but_stob_domain_location;
+
+	c->bc_mkfs_mode = mkfs;
+
+	c->bc_log.lc_got_space_cb = m0_be_engine_got_log_space_cb;
+	c->bc_log.lc_full_cb      = m0_be_engine_full_log_cb;
+	c->bc_log.lc_lock         = &dom->bd_engine_lock;
+
+	be_domain_log_cleanup(c->bc_stob_domain_location, &c->bc_log, c->bc_mkfs_mode);
```

Comment:
Looked inside m0_be_log_create() and it already does something similar to be_domain_log_cleanup(). Please try to remove this call.

Reply:
So, this tool will first cleanup the log and then resize it?

What about the existing content in the log? All of the content is erased?

Reply:
> So, this tool will first cleanup the log and then resize it?
Yes

> What about the existing content in the log? All of the content is erased?
Before resize , we have to trigger replay like below. Please check test results if it is not clarified.

```
[root@ssc-vm-c-0770 744290]# /home/744290/src/OpenSource/main/cortx/cortx-motr/utils/m0betool be_recovery_run /var/motr/m0d-0x7200000000000001\:0xc/db
[root@ssc-vm-c-0770 744290]# /home/744290/src/OpenSource/main/cortx/cortx-motr/utils/m0betool create_be_log /var/motr/m0d-0x7200000000000001\:0xc/db 1073741824
[root@ssc-vm-c-0770 744290]# ls -lh /var/motr/m0d-0x7200000000000001\:0xc/db-log/o/0\:28                                                -rwx------ 1 root root 1.0G Oct 12 08:09 /var/motr/m0d-0x7200000000000001:0xc/db-log/o/0:28
```

Reply:
>Looked inside m0_be_log_create() and it already does something similar to be_domain_log_cleanup(). Please try to remove this call.
```
 432         // be_domain_log_cleanup(c->bc_stob_domain_location, &c->bc_log, c->bc_mkfs_mode);
    433
    434         rc = m0_be_log_create(m0_be_domain_log(dom), &c->bc_log);
    435
    436         m0_be_log_close(m0_be_domain_log(dom));
```
It is not creating BE log file with new size with above code changes.
I believe, we have to remove and create new log file.

Reply:
>It is not creating BE log file with new size with above code changes.
I believe, we have to remove and create new log file.

Thanks for the clarification. I was wrong.

Reply:
as per importing `be_domain_log_cleanup()`.
Consider to use `m0_be_domain_cleanup_by_location()` which has been already imported.

Reply:
Create log file is failed with below change due ot c->bc_log is empty struct passing to be_domain_log_cleanup( )

```
diff --git a/be/ut/helper.c b/be/ut/helper.c
index 45726c5..3ba1755 100644
--- a/be/ut/helper.c
+++ b/be/ut/helper.c
@@ -423,7 +423,7 @@ m0_be_ut_backend_log_resize(struct m0_be_ut_backend *ut_be,
        c->bc_log.lc_full_cb      = m0_be_engine_full_log_cb;
        c->bc_log.lc_lock         = &dom->bd_engine_lock;

-       be_domain_log_cleanup(c->bc_stob_domain_location, &c->bc_log, c->bc_mkfs_mode);
+       m0_be_domain_cleanup_by_location(c->bc_stob_domain_location);

        rc = m0_be_log_create(m0_be_domain_log(dom), &c->bc_log);

```

Reply:
[1:05 PM] Anatoliy Bilenko
    do you think it's due to the following constants being set inside be_domain_log_cleanup():
​[1:05 PM] Anatoliy Bilenko
    &log_cfg->lc_store_cfg.lsc_stob_id.si_domain_fid,
    log_cfg->lc_store_cfg.lsc_stob_domain_location = location;
    log_cfg->lc_store_cfg.lsc_stob_domain_init_cfg = "directio=true";
 
​[1:05 PM] Anatoliy Bilenko
    ?
​[1:05 PM] Venkateswarlu Payidimarry
    

[3:35 PM] Anatoliy Bilenko


&log_cfg->lc_store_cfg.lsc_stob_id.si_domain_fid,
    log_cfg->lc_store_cfg.lsc_stob_domain_location = location;     log_cfg->lc_store_cfg.lsc_stob_domain_init_cfg = "directio=true";




yes
​[1:06 PM] Venkateswarlu Payidimarry
    this changes are not reflecting to main "cfg"


[Anatoliy]
Let's export `be_domain_log_cleanup()` function then.